fbw-a380x/src/wasm/systems/a380_systems/src/hydraulic/flaps_computer.rs

```diff
@@ -159,24 +196,122 @@ impl SlatFlapControlComputer {
+            (1 | 2, 2)
+                if context.indicated_airspeed().get::<knot>()
+                    > Self::FLRS_CONF2_TO_CONF1F_AIRSPEED_THRESHOLD_KNOTS =>
+            {
+                FlapsConf::Conf1F
+            }
+            (2, 2) if self.flaps_conf == FlapsConf::Conf1F => {
+                if context.indicated_airspeed().get::<knot>()
+                    < Self::FLRS_CONF1F_TO_CONF2_AIRSPEED_THRESHOLD_KNOTS
+                {
+                    FlapsConf::Conf2
+                } else {
+                    FlapsConf::Conf1F
+                }
+            }
+            (2 | 3, 3)
+                if context.indicated_airspeed().get::<knot>()
+                    > Self::FLRS_CONF3_TO_CONF2S_AIRSPEED_THRESHOLD_KNOTS =>
+            {
+                FlapsConf::Conf2S
+            }
+            (3, 3) if self.flaps_conf == FlapsConf::Conf2S => {
+                if context.indicated_airspeed().get::<knot>()
+                    < Self::FLRS_CONF2S_TO_CONF3_AIRSPEED_THRESHOLD_KNOTS
+                {
+                    FlapsConf::Conf3
+                } else {
+                    FlapsConf::Conf2S
+                }
+            }
+            (3 | 4, 4)
+                if context.indicated_airspeed().get::<knot>()
+                    > Self::FLRS_CONFFULL_TO_CONF3_AIRSPEED_THRESHOLD_KNOTS =>
+            {
+                FlapsConf::Conf3
+            }
+            (4, 4) if self.flaps_conf == FlapsConf::Conf3 => {
+                if context.indicated_airspeed().get::<knot>()
+                    < Self::FLRS_CONF3_TO_CONFFULL_AIRSPEED_THRESHOLD_KNOTS
+                {
+                    FlapsConf::ConfFull
+                } else {
+                    FlapsConf::Conf3
+                }
+            }
+            (from, 2) if from != 2 => FlapsConf::from(3),
+            (from, to) if from != to => FlapsConf::from(to + 2),
             (_, _) => self.flaps_conf,
         }
     }
 
+    fn flap_load_relief_active(&self, flaps_handle: &FlapsHandle) -> bool {
+        flaps_handle.position() == 2 && self.flaps_conf != FlapsConf::Conf2
+            || flaps_handle.position() == 3 && self.flaps_conf != FlapsConf::Conf3
+            || flaps_handle.position() == 4 && self.flaps_conf != FlapsConf::ConfFull
+    }
+
+    fn cruise_baulk_active(&self, flaps_handle: &FlapsHandle) -> bool {
+        flaps_handle.position() == 1 && self.flaps_conf == FlapsConf::Conf0
+    }
+
+    fn alpha_speed_lock_active(&self, flaps_handle: &FlapsHandle) -> bool {
+        flaps_handle.position() == 0
+            && (self.flaps_conf == FlapsConf::Conf1 || self.flaps_conf == FlapsConf::Conf1F)
+    }
+
     fn surface_movement_required(demanded_angle: Angle, feedback_angle: Angle) -> bool {
         (demanded_angle - feedback_angle).get::<degree>().abs() > Self::EQUAL_ANGLE_DELTA_DEGREE
     }
 
+    fn angle_of_attack(&self, adirs: &impl AdirsMeasurementOutputs) -> Option<Angle> {
+        [1, 2, 3]
```

Comment:
Maybe add a `TODO` comment here to indicate that this is not the correct logic yet.

Reply:
Thanks! Done